examples/md-flexible/src/domainDecomposition/RegularGridDecomposition.cpp

```diff
@@ -252,8 +271,12 @@ void RegularGridDecomposition::exchangeMigratingParticles(AutoPasType &autoPasCo
   }
 }
 
-void RegularGridDecomposition::reflectParticlesAtBoundaries(AutoPasType &autoPasContainer) {
+void RegularGridDecomposition::reflectParticlesAtBoundaries(AutoPasType &autoPasContainer,
+                                                            ParticlePropertiesLibraryType &particlePropertiesLib) {
+  using autopas::utils::Math::isNear;
   std::array<double, _dimensionCount> reflSkinMin{}, reflSkinMax{};
+  auto functorLJ = autopas::LJFunctor<ParticleType, false, true, autopas::FunctorN3Modes::Both, false, false>(
+      _maxReflectiveSkin * 2., particlePropertiesLib);
```

Comment:
Since md-flexible (for now?) always does MD Simulations with the LJ potential this seems reasonable. @SamNewcome btw did we think about what the implications for the potential energy come from the boundary interactions?

Reply:
I think hard coding is fine for now. Using an LJ functor for reflection becomes nasty with multi-site molecules anyway (and I suppose similarly for other rigid-body-type particles e.g. in DEM), so I was going to switch this to its own kernel anyway.

We never thought of the implications for potential energy from reflective boundaries. I would argue, though, that working around this missing potential energy is beyond the scope of md-flexible. I don't think the calculations are necessarily correct even with e.g. periodic boundaries (e.g. I don't think we add potential energy from global forces)